tests/integration/suite/healthz/apphealthz.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(AppHealthz))
+}
+
+// AppHealthz tests that Dapr responds to healthz requests for the app.
+type AppHealthz struct {
+	healthy atomic.Bool
+	server  http.Server
+	done    chan struct{}
+}
+
+func (a *AppHealthz) Setup(t *testing.T, _ context.Context) []framework.RunDaprdOption {
+	a.healthy.Store(true)
+	a.done = make(chan struct{})
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/foo", func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, http.MethodGet, r.Method)
```

Comment:
Note that code inside a handler is executed in a separate goroutine, so we shouldn't use test assertions here.

Instead, make the handler respond with the desired output, and then check the output in the caller.


```go
	mux.HandleFunc("/foo", func(w http.ResponseWriter, r *http.Request) {
		if a.healthy.Load() {
			w.WriteHeader(http.StatusOK)
		} else {
			w.WriteHeader(http.StatusServiceUnavailable)
		}

        fmt.Fprintf(w, "%s %s", r.Method, r.URL.Path)
	})
```

Then in the caller check that the body is the desired one